lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp

```diff
@@ -0,0 +1,267 @@
+#include "lazy_tensor_core/csrc/ts_backend/ts_node_lowering.h"
+
+#include <torch/csrc/jit/frontend/sugared_value.h>
+#include <torch/jit.h>
+#include <torch/torch.h>
+
+#include "lazy_tensor_core/csrc/compiler/node_lowering.h"
+#include "lazy_tensor_core/csrc/ops/as_strided.h"
+#include "lazy_tensor_core/csrc/ops/as_strided_view_update.h"
+#include "lazy_tensor_core/csrc/ops/constant.h"
+#include "lazy_tensor_core/csrc/ops/constant_pad_nd.h"
+#include "lazy_tensor_core/csrc/ops/device_data.h"
+#include "lazy_tensor_core/csrc/ops/expand.h"
+#include "lazy_tensor_core/csrc/ops/ltc_ops.h"
+#include "lazy_tensor_core/csrc/ops/permute.h"
+#include "lazy_tensor_core/csrc/ops/scalar.h"
+#include "lazy_tensor_core/csrc/ops/unsqueeze.h"
+#include "lazy_tensor_core/csrc/ops/view.h"
+#include "lazy_tensor_core/csrc/ts_backend/ts_computation_client.h"
+#include "lazy_tensor_core/csrc/ts_backend/ts_lowering_context.h"
+#include "lazy_tensors/permutation_util.h"
+
+namespace torch_lazy_tensors {
+namespace compiler {
+
+class TSNodeLowering : public NodeLowering {
+ public:
+  TSNodeLowering(const std::string& name, ts_backend::TSLoweringContext* loctx)
+      : NodeLowering(loctx),
+        function_(loctx ? std::make_shared<torch::jit::GraphFunction>(
+                              name, loctx->graph(), nullptr)
+                        : nullptr) {}
+
+  bool Lower(const ir::Node* node) override {
+    TSOpVector ops = LowerToTS(node);
+    if (ops.empty()) {
+      return false;
+    }
+    LTC_CHECK_EQ(node->num_outputs(), ops.size());
+    for (size_t i = 0; i < ops.size(); ++i) {
+      loctx()->AssignOutputOp(ir::Output(node, i), ops[i]);
+    }
+    return true;
+  }
+
+  lazy_tensors::Shape Infer(const ir::Node* node) override {
```

Comment:
oh man.. 
so, first how confident are you about test coverage of what you have here?
second, what does this look like in the world we can use @bdhirsh's work with meta-kernels?

Reply:
side note- I'm hoping that as we get better meta coverage, we'll be able to compute shapes on the spot for **all** xla ops, and finally de-virtualize `Tensor::sizes()`. That doesn't seem worth doing right now in `pytorch/xla`, but seems a lot more reasonable once meta coverage improves and to do directly on LazyTensor (versus trying to do it to the `XLATensor` subclass that's currently in pytorch/xla right now, which would result in a bunch of redundant work).

Reply:
@bdhirsh Faster dispatch is always a good thing! I'm not too worried about it anyway, we cache inference results, so it's not the end of the world if it's not running at lightspeed.